### ISSUE TITLE
Blurry rendering

### ISSUE THREAD

**Issue by erjott** [Cimbali/pympress#43]
### Setup
- Python 3.6.3 on Arch Linux 4.13.11-1
- Pympress 1.1.0b2, installed with pip from github
- external Monitor (1680x1050) on a laptop (2560x1440)

### Expected behaviour 
Text and images should be sharp on all slides

### Steps to reproduce
Open any PDF to start the presentation

### Problem
On the first slide, everything (full screen slide + 2 smaller previews) is nice and sharp. When I switch to the next slide, it and all subsequent slides are rendered blurry, as if they were upscaled from a smaller resolution. 
I can force a re-render by moving the vertical delimiter between the previews or by switching the screens with S, which makes everything on the current and the next slide appear sharp. The following slides are blurry again.
I suspect that this is some scaling issue where the pre-rendered slides don't use the correct resolution. pympress.log also seems to point towards that direction:
> CRITICAL:pympress.__main__:Uncaught exception:
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/pympress/ui.py", line 1166, in on_configure_win
    p_monitor = self.p_win.get_screen().get_monitor_at_window(self.p_frame_cur.get_parent_window())
TypeError: Argument 1 does not allow None as a value

**Comment by Cimbali** [comment]
The incriminated line in ui.py seems to be [#364](https://github.com/Cimbali/pympress/blob/v1.1.0.b2/pympress/ui.py#L364) in release 1.1.0b2, however your message reports 1166. Regardless, `get_parent_window()` is returning `None`, which it shouldn't.

However your problem seems to be different. It looks like the surfacecache is either not aware of the panes' correct size, or not invalidating the cached slides that have an outdated/wrong size.

These could be related. Is the exception you reported the only one you see in your log file? Or does the same exception happen in `setup_screens()` ? 

To do some further checks, could you update to the current master with pip, and post the log file with debug logging enabled?
 ```
pip install --upgrade --no-deps --force-reinstall git+https://github.com/Cimbali/pympress@master#egg=pympress
pympress --log=debug
```

**Comment by erjott** [comment]
Thanks for your fast response. After some digging, I found the solution: Everything works as expected when I export GDK_SCALE=1 before starting Pympress. Not sure why this has never been an issue before with other applications.

For reference, here are two logs for a 5-page PDF:
[pympress_blurry.log](https://github.com/Cimbali/pympress/files/1509288/pympress_blurry.log)
[pympress_ok_scaled.log](https://github.com/Cimbali/pympress/files/1509292/pympress_ok_scaled.log)
The first one without setting GDK_SCALE (everything is in half resolution), the second one with GDK_SCALE=1.

Unfortunately, I can't reproduce the other error anymore, but I'm quite sure there wasn't anything else in the log at the time.

**Comment by Cimbali** [comment]
Alright, so everything in the cache seems to work as expected, but the widget sizes are divided by 2 if you do not force `GDK_SCALE=1`. This variable is related to GDK (on X11) scaling windows on high-DPI monitors. It does seem like this is a cross-platform issue of course. I can reproduce blurry cached slides, by running with `GDK_SCALE=2`.

Do you have a default value of GDK_SCALE or GDK_DPI_SCALE set?

The other error must have been fixed in between, probably something to do with `on_configure_win` being called before all initializations were done.

**Comment by Cimbali** [comment]
Alright, turns out there are surface options we aren't inheriting when caching.

In the non-blurry case we do:
```
        pb = widget.get_window().create_similar_surface(cairo.CONTENT_COLOR, ww, wh)

        cairo_prerender = cairo.Context(pb)
        page.render_cairo(cairo_prerender, ww, wh, wtype)
```
And in the blurry case:
```
        surface = cairo.ImageSurface(cairo.FORMAT_ARGB32, ww, wh)
        context = cairo.Context(surface)
        page.render_cairo(context, ww, wh, wtype)
        del context
```

Where `surface` or `pb` is the same: a cairo surface we save to cache and display using

```
    cairo_context.set_source_surface(pb, 0, 0)
    cairo_context.paint()
```

I'll fix this by modifying the way we create surfaces for caching.

**Comment by Cimbali** [comment]
Now the surfaces aren't blurry anymore. I must say that graphics embedded in pages when the scaling is enabled do not look great. Not sure this can be fixed without a lot of gymnastics though (*), so postponing as low priority for now.

* I'm thinking rendering on target size, i.e. widget size * scaling, while also scaling font sizes and all other tricky things like that, then downscaling to widget size.

**Comment by erjott** [comment]
Thanks for the fix, I can confirm that it works here as well.
